.github/workflows/deploy-docker-compose.yml

```diff
@@ -0,0 +1,161 @@
+      main-image-name:
+        type: string
+        description: "The name of the main image for checking if it exists with the given tag"
+        required: true
+      image-tag:
+        type: string
+        description: "Image tag to deploy (default: pr-<number> if PR exists, latest for default branch)"
+      env-vars:
+        type: string
+        description: "Additional environment variables in KEY=VALUE format, separated by newlines"
+        required: false
+
+jobs:
+  prepare-deploy:
+    runs-on: ubuntu-latest
+    outputs:
+      image-tag-to-deploy: ${{ steps.retrieve-image-tag.outputs.image-tag-to-deploy }}
+    
+    steps:
+      - name: Retrieve image tag to deploy
+        id: retrieve-image-tag
+        run: |
+          if [ -n "${{ inputs.image-tag }}" ]; then
+            echo "Use image-tag input ${{ inputs.image-tag }} for deployment."
+            echo "image-tag-to-deploy=${{ inputs.image-tag }}" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+
+          REF=$(echo "${{ github.event.ref }}" | sed -n 's#^refs/heads/##p')
+          echo "Check if $REF is the default branch (${{ github.event.repository.default_branch }})."
+          if [ "$REF" = "${{ github.event.repository.default_branch }}" ]; then
+            echo "Use latest for deployment."
+            echo "image-tag-to-deploy=latest" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+
+          PULLS=$(curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" "https://api.github.com/repos/${{ github.repository }}/pulls?head=${{ github.repository_owner }}:${REF}")
+          PR_NUMBER=$(echo "$PULLS" | jq -r '.[0].number')
+
+          if [ -z "$PR_NUMBER" ]; then
+            echo "No PR found for branch $REF."
+            exit 1
+          else
+            echo "PR #$PR_NUMBER found for branch $REF."
+            echo "Use image-tag pr-$PR_NUMBER for deployment."
+            echo "image-tag-to-deploy=pr-$PR_NUMBER" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Check if image exists
+        run: |
+          IMAGE_NAME="${{ inputs.main-image-name }}"
+          IMAGE_TAG="${{ steps.retrieve-image-tag.outputs.image-tag-to-deploy }}"
+
+          ENCODED_TOKEN=$(echo -n "${{ secrets.GITHUB_TOKEN }}" | base64)
+          TAG_EXISTS=$(curl -s -H "Authorization: Bearer ${ENCODED_TOKEN}" \
+            https://ghcr.io/v2/${IMAGE_NAME}/tags/list \
+            | jq -r --arg TAG "${IMAGE_TAG}" '.tags[] | select(. == $TAG)')
+          
+          if [ -z "$TAG_EXISTS" ]; then
+            echo "Image ${IMAGE_NAME}:${IMAGE_TAG} does not exist."
+            exit 1
+          else
+            echo "Image ${IMAGE_NAME}:${IMAGE_TAG} exists."
+          fi
+
+  deploy:
+    needs: prepare-deploy
+    runs-on: ubuntu-latest
+    environment: 
+      name: ${{ inputs.environment }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: SSH to VM and execute docker compose down (if exists)
```

Comment:
1) It would be good to have an option to remove volumes with `-v` flag defined in [docs](https://docs.docker.com/reference/cli/docker/compose/down/#options).

And we can get this as an input with default value being false


2) Also with this being added, we can change the `script` section of this task to not use if/elses but dynamically creating the docker compose command maybe?